src/Upgrader/Business/DataProvider/Client/ReleaseApp/ReleaseAppClient.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace Upgrader\Business\DataProvider\Client\ReleaseApp;
+
+use Upgrader\Business\DataProvider\Client\ReleaseApp\Http\HttpClientInterface;
+use Upgrader\Business\DataProvider\Client\ReleaseApp\Http\Request\UpgradeAnalysisRequest;
+use Upgrader\Business\DataProvider\Client\ReleaseApp\Http\Request\UpgradeInstructionsRequest;
+use Upgrader\Business\DataProvider\Client\ReleaseApp\Http\Response\UpgradeAnalysis\Collection\UpgradeAnalysisModuleVersionCollection;
+use Upgrader\Business\DataProvider\Client\ReleaseApp\Http\Response\UpgradeInstructions\Collection\UpgradeInstructionsReleaseGroupCollection;
+use Upgrader\Business\DataProvider\Client\ReleaseApp\Http\Response\UpgradeInstructions\UpgradeInstructionsReleaseGroup;
+use Upgrader\Business\DataProvider\Entity\Collection\ModuleCollection;
+use Upgrader\Business\DataProvider\Entity\Collection\ReleaseGroupCollection;
+use Upgrader\Business\DataProvider\Entity\Module;
+use Upgrader\Business\DataProvider\Entity\ReleaseGroup;
+use Upgrader\Business\DataProvider\Request\DataProviderRequestInterface;
+use Upgrader\Business\DataProvider\Response\DataProviderResponse;
+
+class ReleaseAppClient implements ReleaseAppClientInterface
+{
+    /**
+     * @var \Upgrader\Business\DataProvider\Client\ReleaseApp\Http\HttpClientInterface
+     */
+    protected $httpCommunicator;
```

Comment:
maybe better naming could be RequestExecutor::execute(RequestInterface $request)

Reply:
renamed to $httpClient